dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/Util.java

```diff
@@ -389,6 +438,24 @@ else if (suppressedIgnores != null)
         }
     }
 
+    /**
+     * List of class names of valid return types for resource accessor mtehods.
+     * Enclosed in brackets and delimited by comma.
+     *
+     * @param producer producer of the repository bean, from which it can be
+     *                     determined if the repository is stateful or stateless.
+     */
+    @Trivial
+    static String resourceAccessorTypeNames(RepositoryProducer<?> producer) {
+        Set<Class<?>> types = producer.provider().compat //
+                        .resourceAccessorTypes(producer.stateful());
+
+        StringBuilder b = new StringBuilder('[');
+        for (Class<?> type : types)
+            b.append(b.isEmpty() ? "" : ", ").append(type.getSimpleName());
+        return b.append(']').toString();
```

Comment:
```suggestion
        return types.stream()
                    .map(c -> c.getSimpleName)
                    .collect(Collectors.joining(", ", "[", "]"));
```

issue: Since you instantiate the string builder with `'['` the `isEmpty()` check will always be false resulting in a string like, `[, a, b, c...`